src/app/util/Request/Request.js

```diff
@@ -11,22 +11,33 @@
  */
 
 import { getAuthorizationToken } from 'Util/Auth';
+import { getCurrency } from 'Util/Currency/Current';
 import { hash } from './Hash';
 
+const currencyCode = getCurrency();
 const GRAPHQL_URI = '/graphql';
 
 /**
- * Append authorization token to header object
+ * Append authorization token and Customer specific data to header object if sendSessionData is true
  * @param {Object} headers
- * @returns {Object} Headers with appended authorization
+ * @param {boolean} sendSessionData
+ * @returns {Object} Headers with appended Session Data (token and Currency Code)
  */
-const appendTokenToHeaders = (headers) => {
+const appendSessionDataToHeaders = (headers, sendSessionData) => {
```

Comment:
Prefer adding additional function like `processHeaders` and move token and currency processor calls there.